tests/upgrade_path/test_upgrade_path.py

```diff
@@ -53,24 +81,30 @@ def cleanup_prev_images(duthost):
     duthost.shell("sonic_installer cleanup -y", module_ignore_errors=True)
 
 
-def setup_upgrade_test(duthost, localhost, from_image, to_image, tbinfo,
+def setup_upgrade_test(duthost, localhost, from_image, from_image_version, to_image, to_image_version, tbinfo,
                        upgrade_type, modify_reboot_script=None, allow_fail=False):
-    logger.info("Test upgrade path from {} to {}".format(from_image, to_image))
-    cleanup_prev_images(duthost)
-    # Install base image
-    logger.info("Installing {}".format(from_image))
-    try:
-        target_version = install_sonic(duthost, from_image, tbinfo)
-    except RunAnsibleModuleFail as err:
-        migration_err_regexp = r"Traceback.*migrate_sonic_packages.*SonicRuntimeException"
-        msg = err.results['msg'].replace('\n', '')
-        if re.search(migration_err_regexp, msg):
-            logger.info(
-                "Ignore the package migration error when downgrading to from_image")
-            target_version = duthost.shell(
-                "cat /tmp/downloaded-sonic-image-version")['stdout']
-        else:
-            raise err
+    logger.info("Test upgrade path from {} ({}) to {} ({})".format(from_image, from_image_version,
+                                                                   to_image, to_image_version))
+    target_version = None
+    if from_image_version and restore_image_to_first_boot(duthost, from_image_version):
```

Comment:
This change will help in not installing the base OS which may already be installed.

Is it intentional to not make the change to install the target OS on DUT if it is already installed?